File: tic-tac-toe-game.py
# Tic-Tac-Toe game in Python

board = [
    ['-', '-', '-'],
    ['-', '-', '-'],
    ['-', '-', '-']
]

# Function to print the Tic-Tac-Toe board
def print_board():
    for i in range(3):
        for j in range(3):
            print(board[i][j], end=' ')
        print()

# Function to check if the game has ended
def game_ended():
    # Check rows for a win
    for i in range(3):
        if board[i][0] == board[i][1] == board[i][2] and board[i][0] != '-':
            return True
    # Check columns for a win
    for j in range(3):
        if board[0][j] == board[1][j] == board[2][j] and board[0][j] != '-':
            return True
    # Check diagonals for a win
    if board[0][0] == board[1][1] == board[2][2] and board[0][0] != '-':
        return True
    if board[0][2] == board[1][1] == board[2][0] and board[0][2] != '-':
        return True
    # Check if the board is full (a tie)
    for i in range(3):
        for j in range(3):
            if board[i][j] == '-':
                return False
    return True

# Function to get the user's move
def get_move(player):
    print(f"Player {player}'s turn.")
    while True:
        row = int(input("Enter row number (1-3): ")) - 1
        col = int(input("Enter column number (1-3): ")) - 1
        if row < 0 or row > 2 or col < 0 or col > 2:
            print("Invalid input. Try again.")
        elif board[row][col] != '-':
            print("That space is already occupied. Try again.")
        else:
            board[row][col] = player
            break

# Main game loop
print("Welcome to Tic-Tac-Toe!")
print_board()
player = 'X'
while not game_ended():
    get_move(player)
    print_board()
    if player == 'X':
        player = 'O'
    else:
        player = 'X'
winner = 'X' if player == 'O' else 'O'
print(f"Player {winner} has won the game!")

File: tic-tac-toe-game.go
package main

import (
    "fmt")

var (
    board [3][3]string
    player = "X"
)

func main() {
    clearBoard()
    fmt.Println("Welcome to Tic-Tac-Toe!")
    printBoard()
    for !isGameOver() {
        makeMove()
        printBoard()
    }
}

func clearBoard() {
    for i := 0; i < 3; i++ {
        for j := 0; j < 3; j++ {
            board[i][j] = "-"
        }
    }
}

func printBoard() {
    fmt.Println("  1 2 3")
    for i := 0; i < 3; i++ {
        fmt.Printf("%d %s\n", i+1, board[i])
    }
}

func makeMove() {
    var row, col int
    fmt.Printf("Player %s, enter row and column (e.g. 1 2): ", player)
    fmt.Scanln(&row, &col)
    if row < 1 || row > 3 || col < 1 || col > 3 {
        fmt.Println("Invalid input, please try again")
        makeMove()
    } else if board[row-1][col-1] != "-" {
        fmt.Println("Position already taken, please try again")
        makeMove()
    } else {
        board[row-1][col-1] = player
        if player == "X" {
            player = "O"
        } else {
            player = "X"
        }
    }
}

func isGameOver() bool {
    // check rows
    for i := 0; i < 3; i++ {
        if board[i][0] != "-" && board[i][0] == board[i][1] && board[i][1] == board[i][2] {
            fmt.Printf("Player %s wins!\n", board[i][0])
            return true
        }
    }
    // check columns
    for j := 0; j < 3; j++ {
        if board[0][j] != "-" && board[0][j] == board[1][j] && board[1][j] == board[2][j] {
            fmt.Printf("Player %s wins!\n", board[0][j])
            return true
        }
    }
    // check diagonals
    if board[0][0] != "-" && board[0][0] == board[1][1] && board[1][1] == board[2][2] {
        fmt.Printf("Player %s wins!\n", board[0][0])
        return true
    }
    if board[0][2] != "-" && board[0][2] == board[1][1] && board[1][1] == board[2][0] {
        fmt.Printf("Player %s wins!\n", board[0][2])
        return true
    }
    // check tie
    for i := 0; i < 3; i++ {
        for j := 0; j < 3; j++ {
            if board[i][j] == "-" {
                return false
            }
        }
    }
    fmt.Println("It's a tie!")
    return true
}
